app/Http/Controllers/Entity/InventoryController.php

```diff
@@ -147,4 +159,45 @@ public function destroy(Campaign $campaign, Entity $entity, Inventory $inventory
                 'entity' => $entity->name
             ]));
     }
+
+    /**
+     * @return \Illuminate\Contracts\View\Factory|\Illuminate\View\View
+     * @throws \Illuminate\Auth\Access\AuthorizationException
+     */
+    public function copy(Campaign $campaign, Entity $entity)
+    {
+        $this->authorize('update', $entity->child);
+
+        return view('entities.pages.inventory.copy', compact(
+            'campaign',
+            'entity',
+        ));
+    }
+
+    /**
+     */
+    public function copyFrom(CopyInventory $request, Campaign $campaign, Entity $entity)
+    {
+        $this->authorize('update', $entity->child);
+
+        if ($request->ajax()) {
+            return response()->json(['success' => true]);
+        }
+
+        $copyFrom = Entity::where('id', $request->only('copy_from'))->first();
+        foreach ($copyFrom->inventories->toArray() as $old) {
+            unset($old['id']);
```

Comment:
If you don't use toArray but eloquent's object, you can do duplicate() instead, with the new entity id directly